libraries/mbed/targets/hal/TARGET_ARM_SSG/TARGET_MPS2/us_ticker.c

```diff
@@ -0,0 +1,84 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2015 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <stddef.h>
+#include "us_ticker_api.h"
+#include "PeripheralNames.h"
+
+#define US_TICKER_TIMER1      CMSDK_DUALTIMER1
+#define US_TICKER_TIMER2      CMSDK_DUALTIMER2
+#define US_TICKER_TIMER_IRQn DUALTIMER_IRQn
+
+int us_ticker_inited = 0;
+uint32_t return_value = 0;
+uint32_t timer_value = 0;
+int delta = 0;
+
```

Comment:
These can be moved to their various functions. I’ll correct the code and do another pull request.

From: Martin Kojtal [mailto:notifications@github.com]
Sent: 02 July 2015 13:55
To: mbedmicro/mbed
Cc: Peter Conheeney
Subject: Re: [mbed] commit of mps2 code for pull request to main mbed repo (#1219)


In libraries/mbed/targets/hal/TARGET_ARM_SSG/TARGET_MPS2/us_ticker.c<https://github.com/mbedmicro/mbed/pull/1219#discussion_r33774278>:

> + * See the License for the specific language governing permissions and

> + * limitations under the License.

> + */

> +#include <stddef.h>

> +#include "us_ticker_api.h"

> +#include "PeripheralNames.h"

> +

> +#define US_TICKER_TIMER1      CMSDK_DUALTIMER1

> +#define US_TICKER_TIMER2      CMSDK_DUALTIMER2

> +#define US_TICKER_TIMER_IRQn DUALTIMER_IRQn

> +

> +int us_ticker_inited = 0;

> +uint32_t return_value = 0;

> +uint32_t timer_value = 0;

> +int delta = 0;

> +

Why are these 3 variables in the global scope?

+uint32_t return_value = 0;
+uint32_t timer_value = 0;
+int delta = 0;

—
Reply to this email directly or view it on GitHub<https://github.com/mbedmicro/mbed/pull/1219/files#r33774278>.

-- IMPORTANT NOTICE: The contents of this email and any attachments are confidential and may also be privileged. If you are not the intended recipient, please notify the sender immediately and do not disclose the contents to any other person, use it for any purpose, or store or copy the information in any medium. Thank you.

ARM Limited, Registered office 110 Fulbourn Road, Cambridge CB1 9NJ, Registered in England & Wales, Company No: 2557590
ARM Holdings plc, Registered office 110 Fulbourn Road, Cambridge CB1 9NJ, Registered in England & Wales, Company No: 2548782